Makefile

```diff
@@ -0,0 +1,67 @@
+# Setting SHELL to bash allows bash commands to be executed by recipes.
+# Options are set to exit when a recipe line exits non-zero or a piped command fails.
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
+
+MKFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
+PROJECT_PATH := $(patsubst %/,%,$(dir $(MKFILE_PATH)))
+
+##@ General
+
+# The help target prints out all targets with their descriptions organized
+# beneath their categories. The categories are represented by '##@' and the
+# target descriptions by '##'. The awk commands is responsible for reading the
+# entire set of makefiles included in this invocation, looking for lines of the
+# file as xyz: ## something, and then pretty-format the target and help. Then,
+# if there's a line with ##@ something, that gets pretty-printed as a category.
+# More info on the usage of ANSI control characters for terminal formatting:
+# https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters
+# More info on the awk command:
+# http://linuxcommand.org/lc3_adv_awk.php
+
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-30s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+.PHONY: project-path
+project-path: ## Print the project path.
+	@echo $(PROJECT_PATH)
+
+##@ Tools
+
+# go-install-tool will 'go install' any package $2 and install it to $1.
+define go-install-tool
+@[ -f $(1) ] || { \
+set -e ;\
+TMP_DIR=$$(mktemp -d) ;\
+cd $$TMP_DIR ;\
+go mod init tmp ;\
+echo "Downloading $(2)" ;\
+GOBIN=$(PROJECT_PATH)/bin go install $(2) ;\
+rm -rf $$TMP_DIR ;\
+}
+endef
+
+KIND = $(PROJECT_PATH)/bin/kind
+KIND_VERSION = v0.23.0
+$(KIND):
+	$(call go-install-tool,$(KIND),sigs.k8s.io/kind@$(KIND_VERSION))
+
+.PHONY: kind
+kind: $(KIND) ## Download kind locally if necessary.
+
+##@ Development
+
+.PHONY: test
+test: ## Run tests.
+	$(MAKE) test-unit
+	$(MAKE) test-integration
+
+.PHONY: test-unit
+test-unit: ## Run unit tests.
+	go test -tags=unit -v ./...
```

Comment:
In general I don't like using the `-v` by default when it is being run in automation, but as there is very limited logs currently I am okay with this.